### PR TITLE
EDUCATOR-5499: Prevent learners from submitting empty responses (backend)

### PR DESCRIPTION
**Pull request by jansenk**
**TL;DR -** Prevent learners from submitting empty responses, in the django backend

JIRA: [EDUCATOR-5499](https://openedx.atlassian.net/browse/EDUCATOR-5499)

**What changed?**

- added a method to submission_mixin.py that checks:
         - Are there any text responses in this submission?
         - Are there any file upload responses in this submission? (specifically, are there any file_keys?)
- if not, raise an exception that's caught in the xblock handler method and displays an error message to the learner.
- Removed some test specifically checking that you /can/ do this?? We don't allow this and specifically prevent this.
- I wondered what would happen if you disabled both types of input, and found that there was an awkwardly worded error message. I've also included a rewording of some related messages. 

**Developer Checklist**

- [x] Reviewed the [release process](./release_process.md)
- [x] Translations and JS/SASS compiled
- [ ] Bumped version number in [setup.py](../setup.py) and [package.json](../package.json)

**Testing Instructions**

I never figured out how the one learner managed this, so you're gonna have to take off the javascript that prevents you from doing this.

Go to `openassessment/xblock/static/js/lms/oa_response.js::checkSubmissionAbility`. Change the last line from `this.submitEnabled(readyToSubmit);` to `this.submitEnabled(true);`. You should be able to submit an ORA with no content.

**Reviewer Checklist**

Collectively, these should be completed by reviewers of this PR:

- [x] I've done a visual code review
- [x] I've tested the new functionality

FYI: @edx/masters-devs-gta

### CODE REVIEW

**Review by nsprenkle — APPROVED** [review]
LGTM. Have tested locally 👍
